include/beast/websocket/stream.hpp

```diff
@@ -627,6 +627,14 @@ class stream
         return wr_opcode_ == detail::opcode::text;
     }
 
+
+    /// Returns `true` if either the server or the client sent a close frame.
+    bool 
+    closed() const
+    {
+        return wr_close_ || rd_close_;
```

Comment:
Yes, I use the synchronous version of close. But only in the shutdown case, everything else is asynchronous.

the function stream<NextLayer>::close(close_reason const& cr, error_code& ec) asserts just before checking it to set the error code

I did some debugging yesterday, and I'm pretty sure, that the close function wasn't called twice. It was called just after after upgrading the connection, but before transmitting or receiving any data yet, if I remember correctly.